ror-tools-core/src/main/scala/tech/beshu/ror/tools/core/patches/internal/RorPluginDirectory.scala

```diff
@@ -53,23 +57,33 @@ private [patches] class RorPluginDirectory(val esDirectory: EsDirectory) {
     os.copy(from = file, to = rorPath / file.last)
   }
 
-  def isTransportNetty4PresentInRorPluginPath: Boolean = {
-    findTransportNetty4JarIn(rorPath).isDefined
-  }
-
   def findTransportNetty4Jar: Option[Path] = {
     findTransportNetty4JarIn(rorPath)
   }
 
   def readPatchedByRorVersion(): Option[String] = {
-    Option.when(os.exists(patchedByFilePath)) {
-      os.read(patchedByFilePath)
-    }
+    readEsPatchMetadata().map(_.rorVersion)
   }
 
-  def updatePatchedByRorVersion(): Unit = {
-    os.remove(patchedByFilePath, checkExists = false)
-    os.write(patchedByFilePath, readCurrentRorVersion())
+  def readEsPatchMetadata(): Option[EsPatchMetadata] = {
+    if (os.exists(patchMetadataFilePath)) {
+      EsPatchMetadataCodec.decode(readFile(patchMetadataFilePath)) match {
+        case Right(metadata) => Some(metadata)
+        case Left(error) => throw new IllegalStateException(s"Cannot decode ROR patch metadata file: $error")
```

Comment:
Let's add the content of the file to the exception. It will help us with analysing the issues (clients will likely send only the stacktrace)